lib/connectors/mongodb/mongo_rules_parser.rb

```diff
@@ -0,0 +1,80 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License;
+# you may not use this file except in compliance with the Elastic License.
+#
+
+# frozen_string_literal: true
+
+require 'active_support/core_ext/object'
+require 'connectors/base/simple_rules_parser'
+
+module Connectors
+  module MongoDB
+    class MongoRulesParser < Connectors::Base::SimpleRulesParser
+      def parse_rule(rule)
+        field = rule[:field]
+        value = rule[:value]
+        unless value.present?
+          raise "Value is required for field: #{field}"
+        end
+        unless field.present?
+          raise "Field is required for rule: #{rule}"
+        end
```

Comment:
This code can probably go to the base class

Reply:
I am thinking that it's probably connector-specific. As in, we might have sources where such conditions would make sense.